### PR TITLE
frontmatter for ofeps

### LINKED ISSUE #76: Publish OFEPs to a website that’s searchable.
```[tasklist]
### Tasks
- [ ] Publish OFEPs to a GitHub page
- [ ] Automatically publish when an OFEP PR has been merged
- [ ] Extend OFEPs to include useful metadata (e.g. status, component, author)
- [ ] Each OFEP can be linked to directly
```

### PR DESCRIPTION
**Pull request by mihirm21**
## This PR

- adds the useful frontmatter/metadata to the ofeps
- cleaned the files, to work for publishing using sphinx

### Related Issues

Fixes #76 partially

### Notes

Any images to be added to the ofeps should be in the format :  `unlabelled_image` and not ``.

### Follow-up Tasks

- [ ] a linter could be made for such ofep checks.
- [ ] create an issue to label the images as used in above example

### CODE REVIEW

File: 001-OFEP-cloud-native-pattern.md [hunk 1]
```diff
@@ -120,11 +131,12 @@ In order to enable host containers to consume the sidecar then there should be m
...
+
 
 ## Flow
 The below illustration has been updated also to reflect the current thinking around the initialisation flow of the flagging system.
 
 That said there are some learnings from Istio and concerns around side car overhead - namely around upgrading and maintenance. As such it is worth exploring a pattern for rolling or upgrading sidecars, as the implication is that this will force a deployment rollout due to the change on the deployment object ( and other resource types sts/ds)
 
-<img src="images/001-06.png" width="300">
\ No newline at end of file
+![unlabelled_image](images/001-06.png "unlabelled_image")
```
Comment (justinabrahms) [inline 001-OFEP-cloud-native-pattern.md:142]: If landing this is important w/o providing alt text, at least file a ticket for adding alt text for these things.

File: 005-OFEP-provider-hook.md [hunk 2]
```diff
@@ -1,4 +1,12 @@
-## 005-OFEP-provider-hook.md
+---
+date: 2022-07-30
+title: Provider Hook
```
Comment (justinabrahms) [inline 005-OFEP-provider-hook.md:3]:
```suggestion
title: Provider Hooks
```

File: OFEP-metric-hooks.md [hunk 3]
```diff
@@ -1,4 +1,12 @@
-## OFEP-hook-for-metrics
+---
+date: 2023-06-01
+title: Hook for Metrics
```
Comment (justinabrahms) [inline OFEP-metric-hooks.md:3]:
```suggestion
title: Hooks for Metrics
```

File: OFEP-provider-metadata-capability-discovery.md [hunk 4]
```diff
@@ -1,4 +1,12 @@
-## OFEP-provider-metadata-capability-discovery
+---
+date: 2023-04-15
+title: Provider-Metadata-Capability-Discovery
```
Comment (justinabrahms) [inline OFEP-provider-metadata-capability-discovery.md:3]:
```suggestion
title: Capability discovery for providers
```

File: OFEP-provider-metadata-capability-discovery.md [hunk 5]
```diff
@@ -1,4 +1,12 @@
...
+status: Approved
+authors: [Todd Baert]
+tags: [specification, sdk]
+
+---
+# 016-OFEP-provider-metadata-capability-discovery
```
Comment (justinabrahms) [inline OFEP-provider-metadata-capability-discovery.md:9]:
```suggestion
# Capability discovery for providers
```

File: 009-OFEP-add-dispose.md [hunk 6]
```diff
@@ -1,4 +1,12 @@
-## Add dispose functionality to API
+---
+date: 2023-05-19
+title: Dispose Functionality To API
+status: Approved
+authors: [Todd Baert ,Weyert de Boer]
```
Comment (beeme1mr) [inline 009-OFEP-add-dispose.md:5]:
```suggestion
authors: [Todd Baert, Weyert de Boer]
```

**Review by beeme1mr — COMMENTED** [review]
It would be nice to give images a brief description. If you can't determine the original intent, please create a follow-up issue to address later.

Since you're already modifying most of the existing OFEPs, it may be a good opportunity to remove the number prefixes and move them under a dedicated folder.

File: OFEP-template.md [hunk 7]
```diff
@@ -1,4 +1,12 @@
-## NAME
+---
+date: 2022-07-21
+title: OFEP Template
+status: Active
+authors: [Alex Jones]
```
Comment (beeme1mr) [inline OFEP-template.md:5]:
```suggestion
authors: []
```

File: OFEP-template.md [hunk 8]
```diff
@@ -1,4 +1,12 @@
-## NAME
+---
+date: 2022-07-21
```
Comment (beeme1mr) [inline OFEP-template.md:2]:
```suggestion
date: YYYY-MM-DD
```

File: OFEP-template.md [hunk 9]
```diff
@@ -1,4 +1,12 @@
...
+title: OFEP Template
+status: Active
```
Comment (beeme1mr) [inline OFEP-template.md:4]: What are the allowed statuses?

File: OFEP-template.md [hunk 10]
```diff
@@ -1,4 +1,12 @@
...
+authors: [Alex Jones]
+tags: [template]
```
Comment (beeme1mr) [inline OFEP-template.md:6]:
```suggestion
tags: []
```

File: OFEP/CUE-upstream.md [hunk 11]
```diff
@@ -13,7 +21,8 @@ My proposal here would be to choose a new top-level DSL to define the specificat
 This would mean we could use a single file(s) and toolchain to generate all of the automatically created code for flagD.
 The benefit here would be a single source of truth, ease of contribution, ease of extension and simplified build process.
 
-<img src="images/003-01.png" width="300">
+![unlabelled_image](images/003-01.png "unlabelled_image")
```
Comment (beeme1mr) [inline OFEP/CUE-upstream.md:24]: You'll need to move the images now that you've moved the OFEPs

File: OFEP/provider-hook.md [hunk 12]
```diff
@@ -1,4 +1,12 @@
-## 005-OFEP-provider-hook.md
+---
+date: 2022-07-30
+title: Provider HookS
+status: Approved
+authors: [Shantanu Sen]
+tags: [sdk]
```
Comment (beeme1mr) [inline OFEP/provider-hook.md:6]:
```suggestion
tags: [sdk, spec]
```

File: OFEP/metric-hooks.md [hunk 13]
```diff
@@ -1,4 +1,12 @@
-## OFEP-hook-for-metrics
+---
+date: 2023-06-01
+title: HookS for Metrics
+status: Approved
+authors: [Michael Beemer, Todd Baert, Kavindu Dodanduwa, Justin Abrahms, Giovanni Liva]
+tags: [specification, sdk]
```
Comment (beeme1mr) [inline OFEP/metric-hooks.md:6]:
```suggestion
tags: [spec, sdk]
```

**Comment by beeme1mr** [comment]
Please also update the readme to include these changes.

https://github.com/open-feature/ofep#submitting-the-ofep-and-life-cycle

File: OFEP-template.md [hunk 14]
```diff
@@ -1,4 +1,15 @@
-## NAME
+---
+date: YYYY-MM-DD
+title: OFEP Template
+status: Approved
+authors: []
+tags: []
+<!--- 
+The available status values are : Approved, Pending Review, Rejected, Withdrawn 
+-->
```
Comment (beeme1mr) [inline OFEP-template.md:9]:
```suggestion
<!--- 
Valid statuses: Approved, Rejected, Withdrawn 
-->
status: Approved
authors: []
tags: []
```

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**